freqtrade/strategy/interface.py

```diff
@@ -530,15 +535,22 @@ def get_signal(
             )
             return False, False, None
 
-        enter = latest[SignalType.BUY.value] == 1
+        (enter_type, enter_tag) = (
+            (SignalType.SHORT, SignalTagType.SHORT_TAG)
+            if is_short else
+            (SignalType.BUY, SignalTagType.BUY_TAG)
+        )
```

Comment:
I wonder how you imagine this to work.

The decision if a trade-entry "is_short" is taken right here - so you have no way to have an `is_short` variable passed into this function that is correct (at least not for trade entries - but it's irrelevant for exits too).

The only thing you will have is a config with "short_allowed" (or whatever the naming is) - which will tell you if you can check for longs and shorts, or only for long (in which case shorts will simply be set to False) - but which way the trade is going to go will be decided by the strategy.

Reply:
What do you mean the decision is taken right here? This method is called in freqtradebot with the lines

```
(buy, sell, buy_tag) = self.strategy.get_signal(
            pair,
            self.strategy.timeframe,
            analyzed_df
        )
 ```
 
 which I plan to edit and use
 
 ```
(buy, sell, buy_tag) = self.strategy.get_signal(
            pair,
            self.strategy.timeframe,
            analyzed_df
        )
(short, exit_short, short_tag) = self.strategy.get_signal(
            pair,
            self.strategy.timeframe,
            analyzed_df
        )
 ```

and since both `buy` and `enter_short` will be the dataframe, then 
`        enter = latest[enter_type.value] == 1` will return different values based on what `enter_type.value` is

A lot of the logic in `freqtradebot` is not the same for shorts, and can't be combined into one method

Reply:
So I tried changing

```
(buy, sell, buy_tag) = self.strategy.get_signal(
            pair,
            self.strategy.timeframe,
            analyzed_df
        )
 ```
 
 to be
 
 ```
 (buy, sell, buy_tag) = self.strategy.get_signal(
            pair,
            self.strategy.timeframe,
            analyzed_df
        )
 ```
 
 but got an error that 4 arguments were given but 3 were expected, which I didn't understand, because with my `strategy.get_signal` method, `is_short` is a parameter

Reply:
That's not a very efficient method to search for signals - as it's doing a lot of work twice without any reason.
you'll see how i've solved that.

Reply:
> That's not a very efficient method to search for signals - as it's doing a lot of work twice without any reason.
> you'll see how i've solved that.

Thanks for doing that, what would you say is left for this PR?

Reply:
i'm working on it ... but there's a lot still missing (excluding trades) - as to properly have the strategy interface, at least backtesting must be working - otherwise we can't know if the new interface is working the way it was designed.